sql/tests: TestRandomSyntaxSQLSmith failed [expected exactly one ordering column for RANGE mode with offset]

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/13990720?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/13990720?buildTab=artifacts#/) on release-23.2 @ [5bd2676637d045c80fb0a283f64f3eb2f4d55f59](https://github.com/cockroachdb/cockroach/commits/5bd2676637d045c80fb0a283f64f3eb2f4d55f59):

Random syntax error:

```
    rsg_test.go:911: Crash detected: server panic: pq: internal error: expected exactly one ordering column for RANGE mode with offset
```
Query:

```
        SELECT
        	variance((-7.449973401848254949E+29):::DECIMAL::INT8) OVER (PARTITION BY "%qtab%q333084".crdb_internal_mvcc_timestamp ORDER BY "%qtab%q333084".crdb_internal_mvcc_timestamp ASC NULLS LAST)::DECIMAL
        		AS col793917,
        	concat_agg("%qtab%q333084".co_l1_6::STRING) OVER (PARTITION BY "%qtab%q333084".co_l1_6, "%qtab%q333084".crdb_internal_mvcc_timestamp ORDER BY "%qtab%q333084".crdb_internal_mvcc_timestamp DESC NULLS FIRST RANGE BETWEEN 0.03243688489267516054:::DECIMAL PRECEDING AND UNBOUNDED FOLLOWING)::STRING
        		AS " col😓793918",
        	0:::OID AS col793919
        FROM
        	defaultdb.public."tablE
        1"@[0] AS "%qtab%q333084"
        GROUP BY
        	"%qtab%q333084".co_l1_6, "%qtab%q333084".tableoid, "%qtab%q333084".crdb_internal_mvcc_timestamp
        HAVING
        	bool_and("%qtab%q333084"."col?1_2"::BOOL)::BOOL
        ORDER BY
        	"%qtab%q333084".tableoid
        LIMIT
        	84:::INT8;
```
Schema:

```
    rsg_test.go:720: To reproduce, use schema:
    rsg_test.go:722: SET sql_safe_updates = false;;
    rsg_test.go:722: SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;;
    rsg_test.go:722: SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;;
    rsg_test.go:722: CREATE TABLE "tablE
        1" ("co l1_ 0" INT2 NOT NULL, ")
c͐ol%p1_1" CHAR NOT NULL, "col?1_2" BOOL NOT NULL, col1_3 GEOMETRY, "c'ol1_4" FLOAT4 NULL, col1_5 TSQUERY NOT NULL, co_l1_6 STRING NOT NULL AS (lower(CAST(col1_5 AS STRING))) STORED, "co�l1_7" INT8 NOT NULL AS ("co l1_ 0" + 29543:::INT8) STORED, PRIMARY KEY (")
c͐ol%p1_1" ASC, "co l1_ 0" DESC), INDEX ("c'ol1_4" DESC) WHERE (("tablE
        1".")
c͐ol%p1_1" <= e'\'':::STRING) OR ("tablE
        1"."co l1_ 0" != (-1):::INT8)) OR ("tablE
        1"."co�l1_7" > 127:::INT8));
    rsg_test.go:722: ALTER TABLE "tablE
        1" INJECT STATISTICS e'[{"avg_size": 0, "columns": ["\\"co\\u000bl1_7\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["\\"co l1_ 0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["\\")\\rc\u0350ol%p1_1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_buckets": [{"distinct_range": 0, "num_eq": 319269039465664424, "num_range": 0, "upper_bound": "X"}, {"distinct_range": 399.8890328724099, "num_eq": 7497555065516622757, "num_range": 700, "upper_bound": "``"}], "histo_col_type": "CHAR", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["\\"col?1_2\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col1_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["\\"c\'ol1_4\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "FLOAT4", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col1_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["co_l1_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}]':::JSONB;
    rsg_test.go:722: CREATE TYPE rand_typ_0 AS ENUM ('nma');
    rsg_test.go:722: CREATE TYPE rand_typ_1 AS ENUM ('ono', 'tonbaz', 'hbvrwp', 'cg', 'w');
    rsg_test.go:722: CREATE TYPE rand_typ_2 AS ENUM ('lr', 'e', 'qvl', 'q');
    rsg_test.go:722: SET sql_safe_updates = false;;
    rsg_test.go:722: 
        BEGIN; CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning'); COMMIT;
        BEGIN;
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        COMMIT;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:724: 
    rsg_test.go:725: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith2596237073
--- FAIL: TestRandomSyntaxSQLSmith (384.46s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-36077

--- a/pkg/sql/opt/optbuilder/testdata/window
+++ b/pkg/sql/opt/optbuilder/testdata/window
@@ -1545,3 +1545,14 @@ build
 SELECT avg(k) OVER (ORDER BY id NULLS LAST RANGE 0 PRECEDING) FROM nulls_last_test
 ----
 error: NULLS LAST with RANGE mode with OFFSET is currently unsupported
+
+# Regression test for incorrectly using temporary nulls-last-handling column for
+# first window function in the second window function (#119188).
+build
+SELECT
+  variance(1) OVER (PARTITION BY id ORDER BY id ASC NULLS LAST),
+  min(v) OVER (PARTITION BY v, id ORDER BY id DESC NULLS FIRST RANGE BETWEEN 1 PRECEDING AND UNBOUNDED FOLLOWING)
+FROM
+  nulls_last_test;
+----
+error: NULLS LAST with RANGE mode with OFFSET is currently unsupported

--- a/pkg/sql/opt/optbuilder/window.go
+++ b/pkg/sql/opt/optbuilder/window.go
@@ -429,16 +429,20 @@ func (b *Builder) buildWindowOrdering(
 		cols := flattenTuples([]tree.TypedExpr{te})
 
 		nullsDefaultOrder := b.hasDefaultNullsOrder(t)
+		if !nullsDefaultOrder && isRangeModeWithOffsets {
+			// TODO(#94032): teach the execution engine to support this special
+			// case.
+			panic(errors.New("NULLS LAST with RANGE mode with OFFSET is currently unsupported"))
+		}
 		for k, e := range cols {
 			if !nullsDefaultOrder {
 				expr := tree.NewTypedIsNullExpr(e)
+				// TODO(#94032): reusing an existing column for the temporary IS
+				// NULL expression can be incorrect if that column was created
+				// for the previous window function (perhaps it's incorrect only
+				// when different PARTITION BY clauses are specified).
 				col := outScope.findExistingCol(expr, false /* allowSideEffects */)
 				if col == nil {
-					if isRangeModeWithOffsets {
-						// TODO(yuzefovich): teach the execution engine to
-						// support this special case (#94032).
-						panic(errors.New("NULLS LAST with RANGE mode with OFFSET is currently unsupported"))
-					}
 					// Use an anonymous name because the column cannot be referenced
 					// in other expressions.
 					colName := scopeColName("").WithMetadataName(

